### PR TITLE
Remove throw(Exception) from function definition

### PR DESCRIPTION
**Pull request by teams2ua**
Remove a lot of warnings and prepare for C++17 (throw(Something) is removed from C++)

### CODE REVIEW

File: core/src/utils/DerivationPath.cpp [hunk 1]
```diff
@@ -115,12 +115,12 @@ namespace ledger {
             return (uint32_t) _path.size();
         }
 
-        uint32_t DerivationPath::getLastChildNum() const throw(Exception) {
+        uint32_t DerivationPath::getLastChildNum() const{
```
Comment (phaazon) [inline core/src/utils/DerivationPath.cpp]: Missing space before {.

File: core/src/utils/DerivationPath.cpp [hunk 2]
```diff
@@ -50,7 +50,7 @@ namespace ledger {
 
         }
 
-        std::vector<uint32_t> DerivationPath::parse(const std::string &path) throw(Exception) {
+        std::vector<uint32_t> DerivationPath::parse(const std::string &path){
```
Comment (phaazon) [inline core/src/utils/DerivationPath.cpp]: Missing space before {.

File: core/src/utils/DerivationPath.cpp (hunk 1 above)
Comment (teams2ua) [reply core/src/utils/DerivationPath.cpp]: added

File: core/src/utils/DerivationPath.cpp (hunk 2 above)
Comment (teams2ua) [reply core/src/utils/DerivationPath.cpp]: added

**Pull request merged**